PostHog/PostHogApi.swift

```diff
@@ -279,3 +279,27 @@ class PostHogApi {
         task.resume()
     }
 }
+
+extension PostHogApi {
+    static var jsonDecoder: JSONDecoder = {
+        let decoder = JSONDecoder()
+
+        let dateFormatter = DateFormatter()
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        dateFormatter.timeZone = TimeZone(abbreviation: "UTC")
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
+
+        decoder.dateDecodingStrategy = .custom { decoder in
+            let container = try decoder.singleValueContainer()
+            let dateString = try container.decode(String.self)
+            guard let date = dateFormatter.date(from: dateString) else {
+                throw DecodingError.dataCorruptedError(
+                    in: container, debugDescription: "Invalid date format"
+                )
+            }
+            return date
+        }
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
```

Comment:
Does the API return in the snake case? we need to be sure this is consistent, if there are mixed strategies, this can fail.

Reply:
what `convertFromSnakeCase` do i know, what I meant is that we cannot guarantee that all fields we are going to use are following this convention (since FE and BE have legacy code in different naming strategies, maybe its not an issue for surveys though), but as long as you check and all fields are correctly serializable, then all fine.